### PR TITLE
MusPlayingInfo ZScript Exposure

### PR DESCRIPTION
**Pull request by MajorCooke**
Exposed MusPlayingInfo to ZScript.

- Allows grabbing the currently playing song, base order, and loop properties under the global `mus_playing` named struct.

### CODE REVIEW

File: src/s_sound.cpp [hunk 1]
```diff
@@ -143,6 +143,11 @@ static FString	 LastSong;			// last music that was played
 static FPlayList *PlayList;
 static int		RestartEvictionsAt;	// do not restart evicted channels before this time
 
+DEFINE_GLOBAL(mus_playing);
+DEFINE_FIELD_X(MusPlayingInfo, MusPlayingInfo, name);
```
Comment (alexey-lysiuk) [inline src/s_sound.cpp]: Why not `DEFINE_FIELD` macro?

File: wadsrc/static/zscript/base.zs [hunk 2]
```diff
@@ -42,6 +42,7 @@ struct _ native	// These are the global variables, the struct is only here to av
 	native readonly Weapon WP_NOCHANGE;
 	deprecated("3.8") native readonly bool globalfreeze;
 	native int LocalViewPitch;
+	native readonly @MusPlayingInfo mus_playing;
```
Comment (alexey-lysiuk) [inline wadsrc/static/zscript/base.zs]: I would say that `MusPlaying` or `musplaying` better suits majority of names.

File: src/s_sound.cpp (hunk 1 above)
Comment (alexey-lysiuk) [inline src/s_sound.cpp]: Why not `DEFINE_FIELD` macro?

**Comment by MajorCooke** [comment]
I cannot use DEFINE_FIELD if I have to rename the class on ZScript's side. Just a heads up.

Let's see what Graf says.

**Comment by alexey-lysiuk** [comment]
OK, maybe there are some quirks with that indeed, I cannot check at the moment.

**Comment by MajorCooke** [comment]
Furthermore, FLevelLocals somehow magically translates to LevelLocals on ZScripts side. I don't know how it does that.

**Comment by coelckers** [comment]
DEFINE_FIELD strips the first character of the class name. 
Aside from that, who cases what this struct type is named, it's unlikely to be used anywhere literally. For the variable I'd say get rid of the underscore.

**Comment by MajorCooke** [comment]
Done.

**Pull request merged**